engine/runtime/src/main/scala/org/enso/compiler/Compiler.scala

```diff
@@ -1085,10 +1085,14 @@ class Compiler(
             }
           }
         case None =>
-          // We dont have location information, so we just print the message
+          // There is no source section associated with the diagnostics
           var str = fansi.Str()
+          val fileLocation = diagnostic.location match {
+            case Some(_) => fileLocationFromSection(diagnostic.location, source)
+            case None    => source.getPath
+          }
           str ++= fansi
-            .Str(fileLocationFromSection(diagnostic.location, source))
+            .Str(fileLocation)
```

Comment:
Unrelated change. This fixes the compiler diagnostic output when there is no source available. It used to print `path:: error`, i.e., two consecutive colons. Which was weird.